Client/Application/Shortcuts/ActivityShortcutManager.swift

```diff
@@ -0,0 +1,203 @@
+                return Strings.Shortcuts.activityTypeTabDescription
+            case .clearBrowsingHistory:
+                return Strings.Shortcuts.activityTypeClearHistoryDescription
+            case .enableBraveVPN:
+                return Strings.Shortcuts.activityTypeEnableVPNDescription
+            case .openBraveToday:
+                return Strings.Shortcuts.activityTypeBraveTodayDescription
+            case .openPlayList:
+                return Strings.Shortcuts.activityTypeOpenPlaylistDescription
+        }
+    }
+    
+    /// The phrase suggested to the user when they create a shortcut for the activity
+    var suggestedPhrase: String {
+        switch self {
+            case .newTab:
+                return Strings.Shortcuts.activityTypeNewTabSuggestedPhrase
+            case .newPrivateTab:
+                return Strings.Shortcuts.activityTypeNewPrivateTabSuggestedPhrase
+            case .clearBrowsingHistory:
+                return Strings.Shortcuts.activityTypeClearHistorySuggestedPhrase
+            case .enableBraveVPN:
+                return Strings.Shortcuts.activityTypeEnableVPNSuggestedPhrase
+            case .openBraveToday:
+                return Strings.Shortcuts.activityTypeOpenBraveTodaySuggestedPhrase
+            case .openPlayList:
+                return Strings.Shortcuts.activityTypeOpenPlaylistSuggestedPhrase
+        }
+    }
+}
+
+/// Singleton Manager handles creation and action for Activities
+class ActivityShortcutManager: NSObject {
+    
+    /// Custom Intent Types
+    enum IntentType {
+        case openWebsite
+        case openHistory
+        case openBookmarks
+    }
+
+    // MARK: Lifecycle
+    
+    static var shared = ActivityShortcutManager()
+    
+    // MARK: Activity Creation Methods
+    
+    public func createShortcutActivity(type: ActivityType) -> NSUserActivity {
+        let attributes = CSSearchableItemAttributeSet(itemContentType: kUTTypeItem as String)
+        attributes.contentDescription = type.description
+        
+        let activity = NSUserActivity(activityType: type.identifier)
+        activity.persistentIdentifier = NSUserActivityPersistentIdentifier(type.identifier)
+        
+        activity.isEligibleForSearch = true
+        activity.isEligibleForPrediction = true
+        
+        activity.title = type.title
+        activity.suggestedInvocationPhrase = type.suggestedPhrase
+        activity.contentAttributeSet = attributes
+
+        return activity
+    }
+
+    // MARK: Activity Action Methods
+
+    public func performShortcutActivity(type: ActivityType, using bvc: BrowserViewController) {
+        switch type {
+            case .newTab:
+                bvc.openBlankNewTab(attemptLocationFieldFocus: true, isPrivate: false)
+            case .newPrivateTab:
+                bvc.openBlankNewTab(attemptLocationFieldFocus: true, isPrivate: true)
+            case .clearBrowsingHistory:
+                History.deleteAll {
```

Comment:
I feel like this does not belong here, maybe better to make new BVC method `clearHistoryAndOpenNewTab` or something

Reply:
Moved Navigational functions into Browser. https://github.com/brave/brave-ios/pull/3579/commits/e1e25adef24913c2fa02c7416101415ae1b23053